src/daemon.rs

```diff
@@ -65,13 +86,59 @@ impl Daemon {
         cas: crate::cas::ContentAddressable,
     ) -> Result<(), ExitError> {
         let (activity_tx, activity_rx) = chan::unbounded();
-        let server = rpc::Server::new(socket_path, activity_tx)?;
+
+        let server = rpc::Server::new(socket_path, activity_tx, self.build_events_tx.clone())?;
         let mut pool = crate::thread::Pool::new();
         pool.spawn("accept-loop", move || {
             server
                 .serve()
                 .expect("failed to serve daemon server endpoint");
         })?;
+        let build_events_rx = self.build_events_rx.clone();
+        let mon_tx = self.mon_tx.clone();
+        pool.spawn("build-loop", move || {
+            let mut project_states: HashMap<NixFile, Event> = HashMap::new();
+            let mut event_listeners: Vec<chan::Sender<Event>> = Vec::new();
+
+            for msg in build_events_rx {
+                debug!("Build loop message"; "message" => ?msg);
```

Comment:
This debug message seems likely to cause spam. It's probably useful for building this feature, but I suggest removing it before this PR is merged.